### DISCUSSION TITLE
strategy

### DISCUSSION THREAD

**Discussion by Shweezzy** [s00d/nuxt-i18n-micro#64] [Q&A] [answered]
Hi, I decided to try your package and was pleasantly surprised. I had some difficulties with the strategy implementation.
Is it possible to achieve the result with routing, for example, the main page (and all pages in the future):
**'/' - ua locale (default)
'/ua' - ua locale (default)
'/ru' - ru locale** 

I am currently getting an error: 
[Vue Router warn]: Alias “/:locale(ua)” and the original record: “/” must have the exact same param named ”locale”

### What should I use in this case
definePageMeta({
	alias: []
});
is there a convenient approach with?
**globalLocaleRoutes**

### below I have attached an example using
definePageMeta({
	alias: ['/:locale(ua)']
});

![image](https://github.com/user-attachments/assets/3f59e3e7-6adb-41cf-9e73-8ea08ae5c4d3)

### without using it I get satisfactory routings, but without '/ua'
definePageMeta({
	alias: []
});
![image](https://github.com/user-attachments/assets/c6540a14-4e18-494b-913b-3c2c160a1f5a)

**Comment by s00d** [comment, marked as answer]
Duplicating the locale is not allowed, either
'/' - ua locale
'/ru' - ru locale
or
'/ua' - ua locale
'/ru' - ru locale

It’s the first time I see something like this being needed. Why might this be necessary?

**Reply by Shweezzy** [reply]
I see, thank you for your answer. This is a request from "the top". 
Do you plan to support your package in the future?

**Reply by s00d** [reply]
In general, you can just hack it a bit and create something similar by configuring locales. For example:

```javascript
i18n: {
    locales: [
      { code: 'en_df', iso: 'en_EN', displayName: 'English' },
      { code: 'en', iso: 'en_EN', displayName: 'English' },
    ],
    defaultLocale: 'en_df',
    includeDefaultLocaleRoute: false,
}
```

You’ll just have to duplicate the translation files, but I think that's not a big problem.

Though, you’ll have to create the locale switcher yourself, as using the standard one will duplicate the language.

Regarding support, yes, I’m already working on it.

**Reply by Shweezzy** [reply]
Yes, it works, thanks for the tip
